include/tvm/script/printer/doc.h

```diff
@@ -158,6 +188,454 @@ class LiteralDoc : public ExprDoc {
+};
+
+/*!
+ * \brief Reference type of IndexDocNode.
+ *
+ * \sa IndexDocNode
+ */
+class IndexDoc : public ExprDoc {
+ public:
+  /*!
+   * \brief Constructor of IndexDoc
+   * \param value The target expression of index access.
+   * \param indices The indices to access.
+   */
+  explicit IndexDoc(ExprDoc value, Array<Doc> indices);
+  TVM_DEFINE_NOTNULLABLE_OBJECT_REF_METHODS(IndexDoc, ExprDoc, IndexDocNode);
+};
+
+/*!
+ * \brief Doc that represents function call.
+ *
+ * \sa CallDoc
+ */
+class CallDocNode : public ExprDocNode {
+ public:
+  /*! \brief The callee of this function call */
+  ExprDoc callee{nullptr};
+  /*! \brief The positional arguments */
+  Array<ExprDoc> args;
+  /*! \brief The keys of keyword arguments */
+  Array<String> kwargs_keys;
+  /*!
+   * \brief The values of keyword arguments.
+   *
+   * The i-th element is the value of the i-th key in `kwargs_keys`.
+   * It must have the same length as `kwargs_keys`.
+   */
+  Array<ExprDoc> kwargs_values;
+
+  void VisitAttrs(AttrVisitor* v) {
+    ExprDocNode::VisitAttrs(v);
+    v->Visit("callee", &callee);
+    v->Visit("args", &args);
+    v->Visit("kwargs_keys", &kwargs_keys);
+    v->Visit("kwargs_values", &kwargs_values);
+  }
+
+  static constexpr const char* _type_key = "script.printer.CallDoc";
+  TVM_DECLARE_FINAL_OBJECT_INFO(CallDocNode, ExprDocNode);
+};
+
+/*!
+ * \brief Reference type of CallDocNode.
+ *
+ * \sa CallDocNode
+ */
+class CallDoc : public ExprDoc {
+ public:
+  /*!
+   * \brief Constructor of CallDoc
+   * \param callee The callee of this function call.
+   * \param args The positional arguments.
+   * \param kwargs_keys Keys of keyword arguments.
+   * \param kwargs_values Values of keyword arguments, must have the same length as `kwargs_keys.
+   */
+  CallDoc(ExprDoc callee, Array<ExprDoc> args, Array<String> kwargs_keys,
+          Array<ExprDoc> kwargs_values);
+  TVM_DEFINE_NOTNULLABLE_OBJECT_REF_METHODS(CallDoc, ExprDoc, CallDocNode);
+};
+
+/*!
+ * \brief Doc that represents operation.
+ *
+ * It can be unary, binary and other special operators (for example,
+ * the if-then-else expression).
+ *
+ * \sa OperationDoc
+ */
+class OperationDocNode : public ExprDocNode {
+ public:
+  enum class Kind : int32_t {
+    // Unary operators
+    kUnaryStart,
+    kUSub,    // -x
+    kInvert,  // ~x
+    kUnaryEnd,
+
+    // Binary operators
+    kBinaryStart,
+    kAdd,       // +
+    kSub,       // -
+    kMult,      // *
+    kDiv,       // /
+    kFloorDiv,  // // in Python
+    kMod,       // % in Python
+    kPow,       // ** in Python
+    kLShift,    // <<
+    kRShift,    // >>
+    kBitAnd,    // &
+    kBitOr,     // |
+    kBitXor,    // ^
+    kLt,        // <
+    kLtE,       // <=
+    kEq,        // ==
+    kNotEq,     // !=
+    kGt,        // >
+    kGtE,       // >=
+    kBinaryEnd,
+
+    // Special
+    kSpecialStart,
+    kIfThenElse,  // <operands[1]> if <operands[0]> else <operands[2]>
+    kSpecialEnd
```

Comment:
Let's explicitly specify their values:

```suggestion
    // Unary operators
    kUnaryStart = 0,
    kUSub = 1,    // -x
    kInvert = 2,  // ~x
    kUnaryEnd = 3,

    // Binary operators
    kBinaryStart,
    kAdd,       // +
    kSub,       // -
    kMult,      // *
    kDiv,       // /
    kFloorDiv,  // // in Python
    kMod,       // % in Python
    kPow,       // ** in Python
    kLShift,    // <<
    kRShift,    // >>
    kBitAnd,    // &
    kBitOr,     // |
    kBitXor,    // ^
    kLt,        // <
    kLtE,       // <=
    kEq,        // ==
    kNotEq,     // !=
    kGt,        // >
    kGtE,       // >=
    kBinaryEnd,

    // Special
    kSpecialStart,
    kIfThenElse,  // <operands[1]> if <operands[0]> else <operands[2]>
    kSpecialEnd
```